client/components/domains/domain-product-price/index.jsx

```diff
@@ -252,9 +274,16 @@ class DomainProductPrice extends Component {
 	}
 }
 
-export default connect( ( state ) => ( {
-	domainsWithPlansOnly: getCurrentUser( state )
-		? currentUserHasFlag( state, DOMAINS_WITH_PLANS_ONLY )
-		: true,
-	isCurrentPlan100YearPlan: getSitePlanSlug( state, getSelectedSiteId( state ) ) === PLAN_100_YEARS,
-} ) )( localize( DomainProductPrice ) );
+export default connect( ( state ) => {
+	const sitePlanSlug = getSitePlanSlug( state, getSelectedSiteId( state ) );
+
+	return {
+		domainsWithPlansOnly: getCurrentUser( state )
+			? currentUserHasFlag( state, DOMAINS_WITH_PLANS_ONLY )
+			: true,
+		isCurrentPlan100YearPlan: sitePlanSlug === PLAN_100_YEARS,
+		isBusinessOrEcommerceMonthlyPlan:
+			( sitePlanSlug === PLAN_BUSINESS_MONTHLY || sitePlanSlug === PLAN_ECOMMERCE_MONTHLY ) &&
+			hasDomainCredit( state, getSelectedSiteId( state ) ),
```

Comment:
Just a note: I'd keep these two variables separate (`isBusinessOrEcommerceMonthlyPlan` and `hasDomainCredit`) to be more clear, or maybe rename `isBusinessOrEcommerceMonthlyPlan` to `isBusinessOrEcommerceMonthlyPlanAndHasDomainCredit` 🙂